src/main/java/com/google/step/youtube/YoutubeInfoScraper.java

```diff
@@ -92,4 +97,13 @@ public YoutubeInfoScraper() {
         }
         return Optional.of(response.getItems());
     }
+
+    private Optional<String> getYoutubeChannelResponse(ChannelListResponse response) {
+        // getItems() return null when no items match the criteria (channelId).
+        if (response.getItems() == null || response.getItems().isEmpty()) {
+            return Optional.empty();
+        }
+        // Since we are only requesting one channel-id, we only get one item back.
```

Comment:
instead of a comment just use an assertion.  see go/tott/465

```
checkState(response.getItems().size() == 1, "We should only be requesting a single channelId but got " + response.getItems().size() + " in response");
```

Where that method is: https://guava.dev/releases/19.0/api/docs/com/google/common/base/Preconditions.html#checkState(boolean,%20java.lang.Object)
use a static import.

Reply:
Added, but why do we want to throw an IllegalStateException when the assertion is false?  Instead of just getting the first element in the list and disregarding the rest of the items in the list.